### PR TITLE
vala: Allow installation into default directories

### PR DESCRIPTION
**Pull request by nirbheek**
For generated headers, vapis, and GIRs.

https://github.com/mesonbuild/meson/pull/1469#issuecomment-292814000

### CODE REVIEW

**Comment by arteymix** [comment]
@nirbheek Is there any settings for `vapidir` and `girdir`?

**Comment by arteymix** [comment]
I can confirm that everything work as expected. I would love to see some `--girdir` and `--vapidir` option though and having them reused in the GNOME module.

**Comment by nirbheek** [comment]
@arteymix Yeah, we should have that. Will you submit a PR?

**Comment by TingPing** [comment]
> +                     install : true,
 +                     install_dir : [true, false, false, true],

This might be the wrong issue to bring this up but shouldn't that be reversed?

This at least seems more clear to me:

```meson
install: [true, false, false, true],
install_dir: ['foo', 'bar']
```

**Comment by arteymix** [comment]
@TingPing 

This was discussed in #1469 and the rationale is very simple:

 - `install` indicate a target-wide installation
 - `install_dir` tells where files get installed or not

File: mesonbuild/backend/ninjabackend.py [hunk 1]
```diff
@@ -1079,11 +1079,20 @@ def generate_vala_compile(self, target, outfile):
             args += ['--vapi', os.path.join('..', target.vala_vapi)]
             valac_outputs.append(vapiname)
             target.outputs += [target.vala_header, target.vala_vapi]
+            # Install header and vapi to default locations if user requests this
+            if len(target.install_dir) > 1 and target.install_dir[1] is True:
+                target.install_dir[1] = self.environment.get_includedir()
+            if len(target.install_dir) > 2 and target.install_dir[2] is True:
+                target.install_dir[2] = os.path.join(self.environment.get_datadir(), 'vala', 'vapi')
+            # Generate GIR if requested
```
Comment (arteymix) [inline mesonbuild/backend/ninjabackend.py]: I did the exact same fixes on my local copy :+1:

**Pull request merged**